src/carlib/sports_car.rs

```diff
@@ -1,63 +1,40 @@
 use crate::carlib::car::Car;
-use crate::carlib::car_concept::CarConcept;
-use crate::carlib::car_impl::CarImpl;
+
+use super::car::UNNAMED_CAR;
 
 // Rust seems to encourage composition over direct inheritance.
 // I like that. 👍
+const DEFAULT_ACCELERATION: f64 = 0.84; // I like fast cars and .84 is double of .42.
 pub struct Sportscar {
-    // A Sportcar is a class that has an implementation
-    // `imp` of a car called `CarImpl`. Makes sense.
-    imp: CarImpl, //Perhaps there is a better name than imp?
+    pub acceleration: f64,
+    pub name: String,
+    pub speed: f64,
```

Comment:
Every car has `acceleration`, `name` and `speed`. If I were to create 100 different car types, I would need to specify these over and over again. So ideally we'd only need to specify it once.

Reply:
Again, for simple cases, you could use macros.